docs/overview/architecture.md

```diff
@@ -0,0 +1,244 @@
+# Vald Architecture <!-- omit in toc -->
+This document describes the high-level architecture design of Vald and explains each component in Vald.
+## Table of Contents <!-- omit in toc -->
+
+- [Overview](#overview)
+- [Data Flow](#data-flow)
+  - [Insert](#insert)
+  - [Search](#search)
+- [Components](#components)
+  - [Vald Filter](#vald-filter)
+    - [Vald Ingress Filter](#vald-ingress-filter)
+    - [Vald Egress Filter](#vald-egress-filter)
+    - [Vald Filter Gateway](#vald-filter-gateway)
+  - [Vald Metadata](#vald-metadata)
+    - [Vald Meta Gateway](#vald-meta-gateway)
+    - [Vald Meta](#vald-meta)
+  - [Vald Backup](#vald-backup)
+    - [Vald Compressor](#vald-compressor)
+    - [Vald Backup Manager](#vald-backup-manager)
+    - [Vald Backup Gateway](#vald-backup-gateway)
+  - [Vald Load Balancing](#vald-load-balancing)
+    - [Vald LB Gateway](#vald-lb-gateway)
+    - [Agent Discoverer](#agent-discoverer)
+  - [Vald Core Engine](#vald-core-engine)
+    - [Vald Agent](#vald-agent)
+    - [Vald Agent Scheduler](#vald-agent-scheduler)
+    - [Vald Index Manager](#vald-index-manager)
+  - [Vald Replication Manager](#vald-replication-manager)
+    - [Vald Replication Manager Agent](#vald-replication-manager-agent)
+    - [Vald Replication Manager Controller](#vald-replication-manager-controller)
+  - [Kubernetes Components](#kubernetes-components)
+    - [Kube-apiserver](#kube-apiserver)
+    - [Custom Resources](#custom-resources)
+
+## Overview
+
+
+
+Vald uses a cloud-native architecture focusing on [Kubernetes](https://kubernetes.io/).
+Some components in Vald use Kubernetes API to control the behavior of distributed vector indexes.
+Before reading this document, you need to have some understanding of the basic idea of cloud-native architecture and Kubernetes.
+
+The below image is Vald's architecture.
+
+<img src="../../design/Vald Future Architecture Overview.svg" />
+
+We will explain this image in the following section.
+
+## Data Flow
+
+This section describes the data flow inside Vald and how Vald's vector indexes are stored.
+This is the most important part for the users to understand Vald.
+
+### Insert
+
+<img src="../../assets/docs/insert_flow.svg" />
+
+When the user inserts data into Vald:
+
+1. Vald Ingress receives the request from the user. The request includes the vector and the vector ID.
+2. Vald Ingress will forward the request to the Vald Filter Gateway to pre-process the request data.
+3. Vald Filter Gateway will forward the request to the user-defined Vald Ingress Filter. After the Vald Ingress Filter received the request, it will perform the pre-processing logic defined by the user, for example, padding the vector to match the vector length in Vald.
```

Comment:
```suggestion
3. Vald Filter Gateway will forward the request to the user-defined Vald Ingress Filter. After the Vald Ingress Filter received the request, it will perform the pre-processing logic defined by the user, for example, padding the vector to match the vector dimensionality in Vald.
```